[Hyunjin Kim] iP

hyunjinkim1112 opened the pull request.

hyunjinkim1112 committed:
Level 0. Rename, Greet, Exit

hyunjinkim1112 committed:
Level 1. Echo

It now echos the command that the user inputs.

hyunjinkim1112 committed:
Level 2. Add, List

Add the ability to store whatever text entered by the user and display them back to the user when requested

hyunjinkim1112 committed:
Level 3. Mark as Done

Add the ability to mark tasks as done. add the ability to change the status back to not done.

hyunjinkim1112 committed:
Level 4. ToDos, Events, Deadlines

ToDos: tasks without any date/time attached to it e.g., visit new theme park
Deadlines: tasks that need to be done before a specific date/time e.g., submit report by 11/10/2019 5pm
Events: tasks that start at a specific date/time and ends at a specific date/time

Rishi7830 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,15 @@
+public class Deadline extends Task {
```
Good job on following the coding standards. Smart Move!

Rishi7830 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,15 @@
+    public String getType() {
+        return "D";
+    }
+    @Override
```
I like how you use the override :)

Rishi7830 commented on src/main/java/Mary.java:
```diff
@@ -0,0 +1,107 @@
+import java.util.Scanner;
+
+public class Mary {
+    public static final int MAX_TASKS = 100;
```
Good job on naming your constants all uppercase, thats good code quality

Rishi7830 commented on src/main/java/Mary.java:
```diff
@@ -0,0 +1,107 @@
+    public static final int MAX_TASKS = 100;
+    public static final String LINE_SEPARATOR = "____________________________________________________________ ";
+
+    public static void printHello() {
```
Good job on using the camel case

Rishi7830 commented on src/main/java/Mary.java:
```diff
@@ -0,0 +1,107 @@
+        System.out.print("Enter command (todo, deadline, event, list, mark, unmark, bye): ");
+    }
+
+    public static void printBye() {
```
Good job on the camel case

Rishi7830 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,17 @@
+public class Task {
+    protected String description;
+    protected boolean isDone = false;
```
Good job on naming your boolean variables sound like booleans

Rishi7830 commented on src/main/java/Mary.java:
```diff
@@ -0,0 +1,107 @@
+
+        public void processCommand(String input) {
+            String[] parts = input.split(" ", 2);
+            String command = parts[0];
```
Good job on using plural form for the collection of objects

Rishi7830 reviewed (commented):
Good job on the code quality

hyunjinkim1112 committed:
automated text UI testing

- added input to input.txt, EXPECTED.TXT
- edited runtest.bat

hyunjinkim1112 committed:
refactor tasksManager into another public class file and create package of ip

- added ip package
- refactored taskManager

hyunjinkim1112 committed:
added handling error features

- added MaryException.java file
- edited TaskManager.java to  incorporate error handling

hyunjinkim1112 committed:
Merge branch 'branch-Level-5'

hyunjinkim1112 committed:
added mary package

hyunjinkim1112 committed:
switched from array to arraylist, added delete command

hyunjinkim1112 committed:
added storage class for saving and loading the data, refactored exception

hyunjinkim1112 committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/mary/TaskManager.java

hyunjinkim1112 committed:
small change in the welcome message

hyunjinkim1112 committed:
added saveTasks for every command

Hackin7 commented on src/main/java/mary/InvalidCommandException.java:
```diff
@@ -0,0 +1,23 @@
+////
+//    public static MaryException missingKeyword(String keyword) {
+//        return new MaryException("Missing keyword: \"" + keyword + "\"");
+//    }
```
Remove commented code

Hackin7 commented on src/main/java/mary/task/Task.java:
```diff
@@ -0,0 +1,104 @@
+    public void markAsDone() {
+        this.isDone = true;
+    }
+    public void markAsNotDone() {
```
leave lines between functions for better readability

Hackin7 commented on src/main/java/mary/task/Task.java:
```diff
@@ -0,0 +1,104 @@
+        return new Event(description, isDone, from, to);
+    }
+
+//    public static Task fromFileFormat(String line) {
```
Remove commented code

Hackin7 commented on src/main/java/mary/Parser.java:
```diff
@@ -0,0 +1,91 @@
+public class Parser {
+    public static void processCommand(String command, TaskManager taskManager) throws InvalidCommandException {
+        String[] parts = command.split(" ", 2);
+//        if (parts.length < 2) {
```
Remove commented code

Hackin7 commented on src/main/java/mary/Storage.java:
```diff
@@ -0,0 +1,62 @@
+    }
+
+    public ArrayList<Task> loadFromFile() throws FileNotFoundException {
+        File f = new File(filePath); // create a File for the given file path
```
some comments are self explanatory, can remove

Hackin7 reviewed (commented):
Honestly, quite well structured and readable code, just remove commented code and various other issues

hyunjinkim1112 committed:
added ui class for more OOP / renamed the taskManager to taskList

hyunjinkim1112 committed:
added find command

hyunjinkim1112 committed:
added javadoc comments

hyunjinkim1112 committed:
added javadoc comments

hyunjinkim1112 committed:
Merge pull request #1 from hyunjinkim1112/branch-Level-9

Branch level 9

hyunjinkim1112 committed:
Merge pull request #2 from hyunjinkim1112/branch-A-JavaDoc

Branch a java doc

hyunjinkim1112 committed:
Updated README.md

hyunjinkim1112 committed:
Reverted README.md (of Duke project) back to the original copy

hyunjinkim1112 committed:
edited docs/README.md for user guide

hyunjinkim1112 committed:
edited README.md to change the text style

hyunjinkim1112 committed:
edited README.md for text styling

hyunjinkim1112 committed:
erased unused statements

hyunjinkim1112 committed:
updated task constructor to fix the bug
